optimum/onnxruntime/modeling_decoder.py

```diff
@@ -648,39 +703,440 @@ class ORTModelForCausalLM(ORTModelDecoder, GenerationMixin):
     )
     def forward(
         self,
-        input_ids: torch.LongTensor = None,
+        input_ids: torch.LongTensor,
         attention_mask: Optional[torch.FloatTensor] = None,
-        past_key_values: Optional[Tuple[Tuple[torch.Tensor]]] = None,
         position_ids: Optional[torch.LongTensor] = None,
+        past_key_values: Optional[Tuple[Tuple[torch.Tensor]]] = None,
```

Comment:
Why this reordering?

Reply:
Follows https://github.com/huggingface/optimum/blob/8c296d36f1f9c795bf2a330c6e62381751ca0ed1/optimum/onnxruntime/base.py#L329-L330, needed for IO Binding https://github.com/huggingface/optimum/blob/8c296d36f1f9c795bf2a330c6e62381751ca0ed1/optimum/onnxruntime/base.py#L389 as the input names are ordered given the forward signature https://github.com/huggingface/optimum/blob/8c296d36f1f9c795bf2a330c6e62381751ca0ed1/optimum/onnxruntime/utils.py#L300